### DISCUSSION TITLE
does PerspectiveWidget support PerspectiveTableProxy for remote streaming data?

### DISCUSSION THREAD

**Discussion by phc27x** [perspective-dev/perspective#1849] [Q&A]
Hi,

I'm trying to follow the instructions https://perspective.finos.org/docs/md/python.html about getting python working with perspective.

It says that PerspectiveWidget has same api as perspective-viewer js. I'm trying to make a PerspectiveWidget instance display data from a tornado websocket stream and it keeps saying i can't use a TableProxy, but then i don't know how to 'attach' the live data to the widget.

Is there any example of streaming data into the jupyter lab widget?

Thanks

P

**Comment by timkpaine** [comment]
In Jupyter you have have an implicit running tornado server provided by the Jupyter application, so you can interact between front and backend perspective instances directly via the `PerspectiveWidget`. There are examples provided in the [notebooks](https://github.com/finos/perspective/tree/master/examples/jupyter-notebooks) section of the examples. 

If you wanted to e.g. subscribe to a websocket with a tornado client and push data into perspective, you would just want to call `widget.update(data_in_correct_format_for_perspective)` inside the websocket handler's `onmessage` callback.
